openedx/openedx-platform
# Different transformers for grading

We've observed the task that recalculates grades failing whenever a staff user is enrolled in the course, but some graded content is hidden on a schedule.

The task will create a header for the hidden content but it'll fail to recalculate grades for normal learners as the default transformers applied will hide it.

```
File "/edx/app/edxapp/edx-platform/lms/djangoapps/instructor_task/tasks_helper/grades.py", line 379, in _user_subsection_grades
--
subsection_grade = course_grade.subsection_grade(subsection_location)
File "/edx/app/edxapp/edx-platform/lms/djangoapps/grades/course_grade.py", line 58, in subsection_grade
return self._get_subsection_grade(self.course_data.effective_structure[subsection_key])
File "/edx/app/edxapp/edx-platform/openedx/core/djangoapps/content/block_structure/block_structure.py", line 448, in __getitem__
return self._block_data_map[usage_key]
KeyError: BlockUsageLocator(CourseLocator(u'edx', u'test', u'2019', None, None), u'sequential', u'8487d6b287da461380b939a699b5ef5e')
```

**Sandbox URL**: TBD - sandbox is being provisioned.

**Testing instructions**:

1. Create a course with a graded content hidden due to scheduling.
1. Make sure the course does not use persistent grades.
1. Enroll at least one user to the course.
1. Generate grade reports.

When using this branch, the reports recalculate all the grades.

**Reviewers**
- [ ] @ormsbee 

- - -
**Settings**
```yaml
PROSPECTUS_PREVIEW_HTPASSWD_PASS: "{{ COMMON_HTPASSWD_PASS }}"
```
